stepsfunction/handler.py

```diff
@@ -361,3 +380,50 @@ def get_s3_resource():
         LOG.info('get_s3_resource initializing new connection')
         S3R = boto3.resource('s3')
     return S3R
+
+
+def ocr_by_tesseract(bucket, key):
+    """Convert PDF to TIF single page, then OCR with tesseract.
+
+    :param str key: key name of PDF page for logging
+    :returns: extracted text as a str
+    """
+    # This gs produces much smaller TIF and more clear at 720
+    # than ImageMagick's 'convert' at similar densities
+    LOG.info('ocr_by_tesseract converting page PDF to JPG: bucket {}, key {}'.format(bucket, key))
+    jpg = extract_jpg_from_pdf(bucket, key)
+    jpgfile = open("/tmp/jpgextracted.jpg", "wb")
+    jpgfile.write(jpg)
+    jpgfile.close()
+    txt = run(f'{TESSERACT_BIN} --tessdata-dir {TESSERACT_DATA} /tmp/jpgextracted.jpg stdout')
+    return txt.decode('utf-8')
+
+
+def run(cmd):
+    """Run a command as a subprocess, return output, log output or errors."""
+    LOG.debug('RUN {}')
+    if isinstance(cmd, str):
+        cmd = cmd.split()
+    t_0 = time()
+    res = subprocess.run(cmd, check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    t_run = time() - t_0
+    if res.returncode == 0:
+        LOG.info('run run_seconds=%s cmd="%s"', t_run, cmd)
+        return res.stdout
+    else:
+        msg = 'run: {}'.format(res)
+        LOG.error(msg)
+        raise RuntimeError(msg)
+
+
+def _cleanup():
+    """Remove S3 page pdf and downloaded tmp file.
```

Comment:
we're not running this on S3, right? just locally? It's fine, just want to understand the intent. Maybe update the comments so they don't confuse me? :-)